src/client/mod.rs

```diff
@@ -216,12 +247,29 @@ impl Client {
         })
     }
 
-    pub fn begin(&mut self, n_metrics: u64) -> io::Result<&mut Client> {
+    pub fn begin(&mut self, n_indoms: u64, n_instances: u64, n_metrics: u64) -> io::Result<&mut Client> {
```

Comment:
I know it isn't possible in the current implementation of the writer to do this as we write tocs ahead of time, but lets explore not requiring the user to specify the number of instances and indoms beforehand.